components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java

```diff
@@ -514,6 +516,42 @@ public static boolean removeUserClaimsFromCache(String userName, UserStoreManage
         return true;
     }
 
+    /**
+     * This method will revoke the authorization codes of user.
+     * @param username          username.
+     * @param userStoreManager  userStoreManager.
+     * @return true if revocation is successfull. Else return false
+     * @throws UserStoreException If an error occurred when revoking codes.
+     */
+    public static boolean revokeAuthzCodes(String username, UserStoreManager userStoreManager)
+            throws UserStoreException {
+
+        String userStoreDomain = UserCoreUtil.getDomainName(userStoreManager.getRealmConfiguration());
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(userStoreManager.getTenantId());
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        authenticatedUser.setUserStoreDomain(userStoreDomain);
+        authenticatedUser.setTenantDomain(tenantDomain);
+        authenticatedUser.setUserName(username);
+
+        List<AuthzCodeDO> authorizationCodes;
+        try {
+            authorizationCodes = OAuthTokenPersistenceFactory.getInstance()
+                    .getAuthorizationCodeDAO().getAuthorizationCodesDataByUser(authenticatedUser);
+            for (AuthzCodeDO authorizationCode : authorizationCodes) {
+                OAuthCache.getInstance().clearCacheEntry(new OAuthCacheKey(
+                        OAuth2Util.buildCacheKeyStringForAuthzCode(authorizationCode.getConsumerKey(),
+                                authorizationCode.getAuthorizationCode())));
+                OAuthTokenPersistenceFactory.getInstance().getAuthorizationCodeDAO()
+                        .updateAuthorizationCodeState(authorizationCode.getAuthorizationCode(),
+                                OAuthConstants.AuthorizationCodeState.REVOKED);
+            }
+        } catch (IdentityOAuth2Exception e) {
+            throw new RuntimeException(e);
```

Comment:
This is not ideal, as it could mask the underlying issue and make it difficult to diagnose. It would be better to log the exception and re-throw a more specific exception that accurately reflects the issue.

Reply:
This is missed. Will update.